### PR TITLE
(SIMP-MAINT) Fix OS support data type

### PR DESCRIPTION
**Pull request by trevor-vaughan**
Fixed:
  - Allow the operatingsystemrelease to be optionally defined in
    metadata.json operatingsystem_support

### CODE REVIEW

**Review by op-ct — APPROVED** [review]
:eye: 👍

**Comment by lnemsick-simp** [comment]
spec tests are failing

**Comment by lnemsick-simp** [comment]
This added Amazon Linux to the supported OS's.  How do we know that this is working on Amazon Linux?

File: types/puppet/metadata/os_support.pp [hunk 1]
```diff
@@ -1,5 +1,5 @@
 # The 'operating_support' data structure in metadata.json
 type Simplib::Puppet::Metadata::OS_support = Struct[{
   'operatingsystem'        => String,
-  'operatingsystemrelease' => Array[String]
+  'operatingsystemrelease' => Optional[Array[String]]
```
Comment (lnemsick-simp) [inline types/puppet/metadata/os_support.pp:4]: I don't know why you made this change.  The original changelog entry didn't make sense to me so I updated it to indicate the type change.

**Review by op-ct — APPROVED** [review]
:eye::+1:

**Pull request merged**